docs/pages/core-concepts.mdx

```diff
@@ -0,0 +1,219 @@
+
+Read our [guide to Teleport roles](./access-controls/guides/role-templates.mdx).
+
+## Teleport Auth Service
+
+The **Teleport Auth Service** manages **certificate authorities**, **local
+users**, and **configuration resources** within a **Teleport cluster**. It uses
+storage backends and—if configured—a hardware security module to store the
+certificate authorities' private keys. The Auth Service issues certificates to
+clients and maintains an audit log.
+
+<Figure
+  align="center"
+  bordered="false"
+  caption="Auth service"
+>
+
+![Auth service](../img/architecture/auth.png)
+
+</Figure>
+
+The Auth Service is the only service that has to be connected to a backend. All
+other services are stateless and interact with the Auth Service via a gRPC API.
+**Teleport Services** can connect to the Auth Service either directly or via a
+reverse tunnel established by the **Proxy Service**. This means that you can run
+Teleport Services and the Auth Service in a private network, with only the Proxy
+Service's HTTPS port exposed to the internet.
+
+You can run multiple Auth Service instances in a cluster for high availability.
+
+Read our guides to how [authorization](./architecture/authorization.mdx) and
+[authentication](./architecture/authentication.mdx) work in Teleport.
+
+## Teleport Cloud
+
+**Teleport Cloud** is a managed deployment of the **Teleport Control Plane**.
+Our team at Teleport handles all tasks related to running the **Auth Service**
+and **Proxy Service**, including upgrades and certificate management. Each
+customer account, known as a **Teleport Cloud tenant**, has its own subdomain of
+`.teleport.sh`, e.g., `mytenant.teleport.sh`.
+
+Read more in our [Teleport Cloud
+guide](./deploy-a-cluster/teleport-cloud/introduction.mdx).
+
+## Teleport Cluster
+
+The key concept of Teleport's architecture is the **cluster**. A Teleport
+cluster consists of the **Teleport Auth Service** and **Teleport Proxy Service**,
+plus several **Teleport Services** that manage traffic to resources within your
+infrastructure, such as Kubernetes clusters and Windows desktops. A Teleport
+cluster also includes **Machine ID** and **Teleport Discover**.
+
+A minimal Teleport Cluster consists of the Teleport Auth Service and Teleport
+Proxy Service. In a demo environment, you can run these two services from a
+single **Teleport Process**.
+
+## Teleport Control Plane
+
+The Control Plane consists of the **Auth Service** and **Proxy Service**, which
+govern cluster-wide access. Other **Teleport Services**, which ensure access to
+specific resources, are not considered part of the Teleport Control Plane.
+
+## Teleport Enterprise
+
+**Teleport Enterprise** is a paid plan that includes additional features for the
+**Teleport Control Plane** and **Teleport Services**, plus a support agreement
+with Teleport.
+
+Read more in our [Teleport Enterprise
+guide](./deploy-a-cluster/teleport-enterprise/introduction.mdx).
```

Comment:
I'm not a fan of the terminology "Teleport Enterprise" because it implies that Teleport Cloud is not enterprise. This is how I think about our different offerings:

- Teleport Community Edition (this is open source, free, and self-hosted only)
- Teleport Enterprise Edition (this is paid and can be self-hosted or SaaS)
    - Teleport Enterprise Self-Hosted
    - Teleport Enterprise Cloud